File: focusroom-frontend/src/services/taskService.js
import api from './apiService';

export const taskService = {
    createTask: async (taskName, userId) => {
        const response = await api.post('/tasks', { taskName, userId, status: 'PENDING' });
        return response.data;
    },

    getUserTasks: async (userId) => {
        const response = await api.get(`/tasks/user/${userId}`);
        return response.data;
    },

    updateTaskStatus: async (taskId, status) => {
        const response = await api.put(`/tasks/${taskId}/update?status=${status}`);
        return response.data;
    },

    deleteTask: async (taskId) => {
        const response = await api.delete(`/tasks/${taskId}`);
        return response.data;
    },

    getTaskById: async (taskId) => {
        const response = await api.get(`/tasks/${taskId}`);
        return response.data;
    },
};

File: focusroom-backend/src/main/java/com/focusroom/service/TaskService.java
package com.focusroom.service;

import com.focusroom.model.Task;
import com.focusroom.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {

    @Autowired
    private TaskRepository taskRepository;

    public Task createTask(String userId, String taskName) {
        Task task = new Task(userId, taskName);
        return taskRepository.save(task);
    }

    public List<Task> getUserTasks(String userId) {
        return taskRepository.findByUserIdOrderByCreatedAtDesc(userId);
    }

    public Task updateTaskStatus(String taskId, Task.TaskStatus status) {
        Task task = taskRepository.findById(taskId)
                .orElseThrow(() -> new RuntimeException("Task not found with id: " + taskId));

        task.setStatus(status);
        return taskRepository.save(task);
    }

    public void deleteTask(String taskId) {
        taskRepository.deleteById(taskId);
    }

    public Task getTaskById(String taskId) {
        return taskRepository.findById(taskId)
                .orElseThrow(() -> new RuntimeException("Task not found"));
    }
}
